cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -732,6 +733,17 @@ def make_resource(self, data, **kwargs):
         return Iam(**data)
 
 
+class ImdsSchema(BaseSchema):
+    """Represent the schema of IMDS for HeadNode."""
+
+    secured = fields.Bool(metadata={"update_policy": UpdatePolicy.UNSUPPORTED})
```

Comment:
couldn't update for this be supported?

Reply:
> Is there a way we can add a check for the case where the head node's IMDS is configured to be secured and Batch is used as the scheduler, and if so print a warning (or even fail config validation)?

Sure, we are going to fail the validation when HeadNode.Imds.Secured is True and Scheduling.Scheduler is awsbatch.
Will add this in the next iteration

Reply:
> couldn't update for this be supported?

Think it's possible. Working on it.

Reply:
> Just to be precise: the update of HeadNode.Imds.Secured implies the update of Cfn resource HeadNodeLaunchTemplate.Metadata.AWS::CloudFormation::Init.cfnHupConfig

would that require the replacement of the instance? it would think not, but i could be wrong.

Reply:
No, it doesn't.